server/games/game.py

```diff
@@ -836,49 +757,20 @@ def sanitize_name(self, name: str) -> str:
 
         # Currently, we can only end up here if a game desynced or was a custom game that terminated
         # too quickly.
-        async with self._db.engine.acquire() as conn:
+        async with self._db.acquire() as conn:
             await conn.execute(
                 "UPDATE game_stats SET validity = %s "
                 "WHERE id = %s", (new_validity_state.value, self.id))
 
     def get_army_score(self, army):
-        """
-        Since we log multiple results from multiple sources, we have to pick one.
-
-        On conflict we try to pick the most frequently reported score. If there
-        are multiple scores with the same number of reports, we pick the greater
-        score.
-
-        TODO: Flag games with conflicting scores for manual review.
-        :param army index of army
-        :raise KeyError
-        :return:
-        """
-        scores: Dict[int, int] = Counter(
-            map(lambda res: res[2], self._results.get(army, []))
-        )
-
-        # There were no results
-        if not scores:
-            return 0
-
-        # All scores agreed
-        if len(scores) == 1:
-            return scores.popitem()[0]
-
-        # Return the highest score with the most votes
-        self._logger.info("Conflicting scores (%s) reported for game %s", scores, self)
-        score, _votes = max(scores.items(), key=lambda kv: kv[::-1])
-        return score
+        return self._results.score(army, self.id)
 
     def get_army_result(self, player):
```

Comment:
Maybe this should be called `get_player_outcome`?

Reply:
It does return a score (a 'score' value from the game) instead of an outcome ('victory', 'defeat' etc.).

Reply:
Sotty, I meant `get_army_result`, which takes a `Player` as argument and returns a `GameOutcome` (not `get_army_score`)

Reply:
I didn't want to touch `Game` class public methods, though I could rename that if you want.

Reply:
It just looked off to me, so I'm all for it. I can do it in my later PR if you don't want to :)